crates/matrix-sdk/src/account.rs

```diff
@@ -921,6 +921,52 @@ impl Account {
                 )
             }))
     }
+
+    /// Retrieves the user's recently visited room list
+    pub async fn get_recently_visited_rooms(&self) -> Result<Vec<String>> {
+        let user_id = self.client.user_id().ok_or(Error::AuthenticationRequired)?;
+        let data = self
+            .client
+            .store()
+            .get_kv_data(StateStoreDataKey::RecentlyVisitedRooms(user_id))
+            .await?;
+
+        Ok(data
+            .map(|v| {
+                v.into_recently_visited_rooms()
+                    .expect("Session data is not a list of recently visited rooms")
+            })
+            .unwrap_or_default())
+    }
+
+    /// Moves/inserts the given room to the front of the recently visited list
+    pub async fn track_recently_visited_room(&self, room_id: String) -> Result<(), Error> {
+        let user_id = self.client.user_id().ok_or(Error::AuthenticationRequired)?;
+
+        if let Err(error) = RoomId::parse(&room_id) {
+            error!("Invalid room id: {}", error);
+            return Err(Error::Identifier(error));
+        }
+
+        // Get the previously stored recently visited rooms
+        let mut recently_visited_rooms = self.get_recently_visited_rooms().await?;
+
+        // Remove all other occurrences of the new room_id
+        recently_visited_rooms.retain(|r| r != &room_id);
+
+        // And insert it as the most recent
+        recently_visited_rooms.insert(0, room_id);
+
+        // Cap the whole list to 20
+        recently_visited_rooms = recently_visited_rooms.iter().take(20).cloned().collect();
```

Comment:
nit: let's use a `const` value for the threshold value of 20, so it's not a magic constant

also, to avoid a potential reallocation:

```suggestion
        recently_visited_rooms.truncate(20);
```